model/DataStore/PersistDataService.php

```diff
@@ -164,4 +145,49 @@ private function getFileSystemManager(): FileSystemService
     {
         return $this->getServiceLocator()->get(FileSystemService::SERVICE_ID);
     }
+
+    private function addMetaDataFile($zipFile, string $fileNameToAdd, string $content): bool
+    {
+        $return = false;
+
+        try {
+            $return = $zipFile->addFromString($fileNameToAdd, $content);
+        } catch (Throwable $exception) {
+            $this->logError(
+                sprintf('Error while adding file %s to the zip file  with message: %s ',
+                    $fileNameToAdd,
+                    $exception->getMessage())
+            );
+        }
+
+        return $return;
+    }
+
+    private function addMetaDataToArchive($zipFile, array $metaData): void
```

Comment:
Should we extract those zip operation to separate service so it will be easier for unit tests?

Reply:
the majority of functions are returning void , 
let me try it and then i will comeback to you 
as the test will not be possible or reliable

Reply:
split done